posthog/api/test/test_person.py

```diff
@@ -599,22 +599,30 @@ def test_csv_export(self):
 
     def test_pagination_limit(self):
         created_ids = []
-        for index in range(0, 20):
+
+        for index in range(0, 19):
             created_ids.append(str(index + 100))
-            _create_person(
+            Person.objects.create(  # creating without _create_person to guarentee created_at ordering
                 team=self.team, distinct_ids=[str(index + 100)], properties={"$browser": "whatever", "$os": "Windows"},
             )
 
-        with freeze_time(datetime(2022, 1, 1, 0, 0)):
-            flush_persons_and_events()
+        # Very occasionally, a person might be deleted in postgres but not in Clickhouse due to network issues or whatever
+        # In this case Clickhouse will return a user that then doesn't get returned by postgres.
+        # We would return an empty "next" url.
+        # Now we just return 9 people instead
+        create_person(team_id=self.team.pk, version=0)
+
         returned_ids = []
-        response = self.client.get("/api/person/?limit=10").json()
-        self.assertEqual(len(response["results"]), 10)
+        with self.assertNumQueries(6):
+            response = self.client.get("/api/person/?limit=10").json()
+        self.assertEqual(len(response["results"]), 9)
         returned_ids += [x["distinct_ids"][0] for x in response["results"]]
         response = self.client.get(response["next"]).json()
         returned_ids += [x["distinct_ids"][0] for x in response["results"]]
+        self.assertEqual(len(response["results"]), 10)
 
-        self.assertCountEqual(returned_ids, created_ids, returned_ids)
+        created_ids.reverse()  # ids are returned in desc order
+        self.assertEqual(returned_ids, created_ids, returned_ids)
```

Comment:
I think this only needs to be returned_ids, created_ids?

Reply:
oh that last one just displays it for debugging if the test fails